sycl/test/fpga_tests/fpga_ac_int.cpp

```diff
@@ -0,0 +1,940 @@
+void test_sub_equal(queue &Queue, const ac_int<W, S> a, const ac_int<W2, S2> b,
+                    ac_int<W3, S3> &c) {
+  buffer<ac_int<W, S>, 1> inp1(&a, 1);
+  buffer<ac_int<W2, S2>, 1> inp2(&b, 1);
+  buffer<ac_int<W3, S3>, 1> result(&c, 1);
+
+  Queue.submit([&](handler &h) {
+    auto x = inp1.template get_access<kSyclRead>(h);
+    auto y = inp2.template get_access<kSyclRead>(h);
+    auto res = result.template get_access<kSyclWrite>(h);
+    h.single_task<class sub_equal>([=]() {
+      res[0] = x[0];
+      res[0] -= y[0];
+    });
+  });
+  Queue.wait();
+}
+
+template <int W, int S, int W2, int S2, int W3, int S3>
+void test_div(queue &Queue, const ac_int<W, S> a, const ac_int<W2, S2> b,
+              ac_int<W3, S3> &c) {
+  buffer<ac_int<W, S>, 1> inp1(&a, 1);
+  buffer<ac_int<W2, S2>, 1> inp2(&b, 1);
+  buffer<ac_int<W3, S3>, 1> result(&c, 1);
+
+  Queue.submit([&](handler &h) {
+    auto x = inp1.template get_access<kSyclRead>(h);
+    auto y = inp2.template get_access<kSyclRead>(h);
+    auto res = result.template get_access<kSyclWrite>(h);
+    h.single_task<class div>([=]() { res[0] = x[0] / y[0]; });
+  });
+  Queue.wait();
+}
+
+template <int W, int S, int W2, int S2, int W3, int S3>
+void test_div_equal(queue &Queue, const ac_int<W, S> a, const ac_int<W2, S2> b,
+                    ac_int<W3, S3> &c) {
+  buffer<ac_int<W, S>, 1> inp1(&a, 1);
+  buffer<ac_int<W2, S2>, 1> inp2(&b, 1);
+  buffer<ac_int<W3, S3>, 1> result(&c, 1);
+
+  Queue.submit([&](handler &h) {
+    auto x = inp1.template get_access<kSyclRead>(h);
+    auto y = inp2.template get_access<kSyclRead>(h);
+    auto res = result.template get_access<kSyclWrite>(h);
+    h.single_task<class div_equal>([=]() {
+      res[0] = x[0];
+      res[0] /= y[0];
+    });
+  });
+  Queue.wait();
+}
+
+template <int W, int S, int W2, int S2, int W3, int S3>
+void test_mult(queue &Queue, const ac_int<W, S> a, const ac_int<W2, S2> b,
+               ac_int<W3, S3> &c) {
+  buffer<ac_int<W, S>, 1> inp1(&a, 1);
+  buffer<ac_int<W2, S2>, 1> inp2(&b, 1);
+  buffer<ac_int<W3, S3>, 1> result(&c, 1);
+
+  Queue.submit([&](handler &h) {
+    auto x = inp1.template get_access<kSyclRead>(h);
+    auto y = inp2.template get_access<kSyclRead>(h);
+    auto res = result.template get_access<kSyclWrite>(h);
+    h.single_task<class mult>([=]() { res[0] = x[0] * y[0]; });
+  });
+  Queue.wait();
+}
+
+template <int W, int S, int W2, int S2, int W3, int S3>
+void test_mult_equal(queue &Queue, const ac_int<W, S> a, const ac_int<W2, S2> b,
+                     ac_int<W3, S3> &c) {
+  buffer<ac_int<W, S>, 1> inp1(&a, 1);
+  buffer<ac_int<W2, S2>, 1> inp2(&b, 1);
+  buffer<ac_int<W3, S3>, 1> result(&c, 1);
+
+  Queue.submit([&](handler &h) {
+    auto x = inp1.template get_access<kSyclRead>(h);
+    auto y = inp2.template get_access<kSyclRead>(h);
+    auto res = result.template get_access<kSyclWrite>(h);
+    h.single_task<class div_equal>([=]() {
```

Comment:
We can't have two kernels with the same name in the module.

Reply:
Corrected, not sure why my dpcpp run did not catch this. Will look into that as well.